File: C/nth_prime.c
// nth prime number finder without using functions

#include <stdio.h>

int main() {
    int n, count = 0, num = 2, isPrime;

    printf("Enter the value of n: ");
    scanf("%d", &n);

    if (n <= 0) {
        printf("Invalid input\n");
        return 0;
    }

    while (count < n) {
        isPrime = 1;

        for (int i = 2; i * i <= num; i++) {
            if (num % i == 0) {
                isPrime = 0;
                break;
            }
        }

        if (isPrime) {
            count++;
        }

        num++;
    }

    printf("The %dth prime number is %d\n", n, num - 1);
    return 0;
}

File: Python/nth_prime.py
def main():
    n = int(input("Enter the value of n: "))
    count = 0
    num = 2

    if n <= 0:
        print("Invalid input")
        return

    while count < n:
        isPrime = 1
        i = 2
        while i * i <= num:
            if num % i == 0:
                isPrime = 0
                break
            i += 1

        if isPrime:
            count += 1
        num += 1

    print(f"The {n}th prime number is {num - 1}")


if __name__ == "__main__":
    main()

